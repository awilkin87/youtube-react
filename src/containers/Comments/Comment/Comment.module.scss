@use '../../../styles/shared.module';

.container {
  display: flex;
  margin: 16px 0;

  .avatar {
    @include shared.left-avatar;
    flex-shrink: 0;
  }

  .userName {
    font-weight: 600;
    margin-bottom: 4px;
  }

  .actions {
    margin-top: 4px;

    button {
      margin-left: 8px;

      // Override Semantic UI values
      font-weight: 500px;
      text-transform: uppercase;
    }
  }
}
