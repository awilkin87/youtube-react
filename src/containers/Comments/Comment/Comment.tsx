import { Button, Image } from 'semantic-ui-react';
import CommentRating from '../CommentRating/CommentRating';
import styles from './Comment.module.scss';

export interface CommentProps {
  // TODO
}

const Comment: React.FunctionComponent<Readonly<CommentProps>> = function (props) {
  return (
    <div className={styles.container}>
      <Image className={styles.avatar} src='http://via.placeholder.com/48x48' circular />
      <div>
        <div className={styles.userName}>User name</div>
        <span>Comment text</span>
        <div className={styles.actions}>
          <CommentRating likes={1} />
          <Button size='mini' compact>Reply</Button>
        </div>
      </div>
    </div>
  );
}

export default Comment;
