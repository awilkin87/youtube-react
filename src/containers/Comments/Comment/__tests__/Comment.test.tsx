import { shallow } from 'enzyme'
import Comment from '../Comment'

test('Render Comment', () => {
  const wrapper = shallow(<Comment />);
  expect(wrapper).toMatchSnapshot();
});
