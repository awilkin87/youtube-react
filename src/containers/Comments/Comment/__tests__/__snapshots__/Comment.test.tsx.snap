// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Comment 1`] = `
<div
  className="container"
>
  <Image
    as="img"
    circular={true}
    className="avatar"
    src="http://via.placeholder.com/48x48"
    ui={true}
  />
  <div>
    <div
      className="userName"
    >
      User name
    </div>
    <span>
      Comment text
    </span>
    <div
      className="actions"
    >
      <CommentRating
        likes={1}
      />
      <Button
        as="button"
        compact={true}
        size="mini"
      >
        Reply
      </Button>
    </div>
  </div>
</div>
`;
